# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/base/crashtests/1566310.html
# Commit: ce9f99b79f12
# Full Hash: ce9f99b79f12aae2cf231004197b92b3b372e0bc
# Author: Kashav Madan <kmadan@mozilla.com>
# Date: 2019-07-17 09:36:40
# Description:
#   Bug 1566310 - Nullcheck mWindowGlobalChild prior to notifying it of beforeunload listeners, r=nika
#   
#   Missed in the 1543251 refactor.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D38223
# ==============================================================================

diff -r 663f5d1ba3f3 -r ce9f99b79f12 dom/base/crashtests/1566310.html
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/base/crashtests/1566310.html	Tue Jul 16 18:35:25 2019 +0000
@@ -0,0 +1,24 @@
+<!doctype html>
+<html>
+<head>
+<script type="text/javascript">
+function noop() {}
+function crash() {
+  let div = document.querySelector("div");
+  let ifr = document.querySelector("iframe");
+
+  // We need a reference to the iframe's window from *before* it gets detached.
+  let win = ifr.contentWindow;
+
+  div.appendChild(ifr);
+
+  win.addEventListener("beforeunload", noop);
+  win.removeEventListener("beforeunload", noop);
+}
+</script>
+</head>
+<body onload="crash()">
+  <div></div>
+  <iframe></iframe>
+</body>
+</html>